dataland-frontend/src/components/resources/frameworkDataSearch/euTaxonomy/EuTaxonomyPanelNonFinancials.vue

```diff
@@ -77,45 +77,55 @@
   </div>
 </template>
 
-<script>
+<script lang="ts">
 import { ApiClientProvider } from "@/services/ApiClients";
+import { EuTaxonomyDataForNonFinancials } from "@clients/backend";
 import TaxoCard from "@/components/resources/frameworkDataSearch/euTaxonomy/EuTaxoCard.vue";
 import TaxoInfoCard from "@/components/resources/frameworkDataSearch/euTaxonomy/EuTaxoInfoCard.vue";
+import { defineComponent, inject } from "vue";
+import Keycloak from "keycloak-js";
+import { assertDefined } from "@/utils/TypeScriptUtils";
 
-export default {
+export default defineComponent({
   name: "EuTaxonomyPanelNonFinancials",
   components: { TaxoCard, TaxoInfoCard },
   data() {
     return {
-      dataSet: null,
+      dataSet: null as EuTaxonomyDataForNonFinancials | null | undefined,
+    };
+  },
+  setup() {
+    return {
+      getKeycloakPromise: inject<() => Promise<Keycloak>>("getKeycloakPromise"),
     };
   },
   props: {
-    dataID: String,
+    dataID: {
+      type: String,
+    },
   },
   mounted() {
-    this.getCompanyEUDataset();
+    void this.getCompanyEuDataset();
   },
   watch: {
-    dataID() {
-      this.getCompanyEUDataset();
+    async dataID() {
+      await this.getCompanyEuDataset();
```

Comment:
Generell sollten wir uns die einführung von async/await noch mal ansehen

Reply:
man könnte hier auch einfach ein void nehmen das würde eslint auch zufrieden stellen
```
dataID() {
      void this.getCompanyEuDataset();
}
```
so habe ich das an anderen Stellen gemacht

Reply:
tatsächlich konnte an vielen Stellen kein return verwendet werden wegen "Property does not exist on ...". Habe das auf return wo möglich und void ansonsten umgestellt.

Reply:
ok